src/main/java/seedu/address/logic/commands/FavoriteCommand.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Favorites a Person in the Address Book
+ */
+public class FavoriteCommand extends Command {
+
+    public static final String COMMAND_WORD = "fav";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Favourites the person according "
+            + "to the Contact ID provided by user input.\n "
+            + "Parameters: INDEX (must be positive integer).\n"
+            + "Example: " + COMMAND_WORD + "1";
+
+    public static final String MESSAGE_NOT_EXISTS = "User doesn't exist in the address book";
+
+    public static final String MESSAGE_ISFAVORITED = "User is already starred in this address book";
+
+    public static final String MESSAGE_FAVORITE_PERSON_SUCCESS = "Favourited Person: %1$s";
+
+    public final Index index;
+
+    public FavoriteCommand(Index index) {
+        this.index = index;
+    }
+
+    /**
+     * Returns a CommandResult where the Selected Person
+     * is Favorited.
+     */
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_NOT_EXISTS);
+        }
+
+        Person personToFav = lastShownList.get(index.getZeroBased());
+
+        if (personToFav.getIsFavorite().getFavoriteStatus()) {
+            throw new CommandException(MESSAGE_ISFAVORITED);
+        }
```

Comment:
Could consider changing this to a success path, because as an exception the user still has to backspace the `fav` command before typing a new command.

For e.g. can change to:

```suggestion
        if (personToFav.getIsFavorite().getFavoriteStatus()) {
            return new CommandResult(String.format("Person is already a favourite: %s", starredPerson));
        }
```

Reply:
Okay edited for Both